### PR TITLE
Add ssh to machine button to machineview machines

### PR DESCRIPTION
**Pull request by jaycee**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8700/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8701/

File: Makefile [hunk 1]
```diff
@@ -91,7 +91,7 @@ sysdeps:
 # TOOLS
 #######
 $(PY):
-	virtualenv .
+	virtualenv . --python=python2
```
Comment (frankban) [inline Makefile]: TY!

File: jujugui/static/gui/src/app/components/machine-view/machine/machine.js [hunk 2]
```diff
@@ -244,6 +245,24 @@ class MachineViewMachine extends React.Component {
     );
   }
 
+  _sshToMachine() {
+    const machine = this.props.machineAPI.machine.id;
+    const commands = [`juju ssh ${machine}`]
+    this.props.changeState({terminal: commands});
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/machine-view/machine/machine.js]: Ha! It's cool that it's that easy.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8703/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8704/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8705/

File: jujugui/static/gui/src/app/components/machine-view/machine/machine.js [hunk 3]
```diff
@@ -244,6 +245,25 @@ class MachineViewMachine extends React.Component {
...
 
+  _sshToMachine() {
+    const machine = this.props.machineAPI.machine.id;
+    const commands = [`juju ssh ${machine}`];
+    this.props.changeState({terminal: commands});
+  }
+
+  _generateTerminalButton() {
+    const props = this.props;
+    if (props.type != 'container' && props.showSSHButton) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/machine-view/machine/machine.js]: !==

File: jujugui/static/gui/src/app/init/component-renderers-mixin.js [hunk 4]
```diff
@@ -698,6 +704,15 @@ Browser: ${navigator.userAgent}`
     const ecs = modelAPI.get('ecs');
     const decorated = MachineView.DecoratedComponent;
     const propTypes = decorated.propTypes;
+    const showSSHButtons = (
+      this.applicationConfig.flags.terminal ||
```
Comment (hatched) [inline jujugui/static/gui/src/app/init/component-renderers-mixin.js]: Lets create a util fn for this as I believe it's used in multiple places.

**Comment by jaycee** [comment]
Thanks all!

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8710/

**Pull request merged**